### PR TITLE
Profiles

### PR DESCRIPTION
**Pull request by borcherd**
## Summary by CodeRabbit

- **New Features**
	- Added new API routes for user data management, including fetching and updating user information, JWT token management, and profile photo submission.
	- Introduced a profile management interface with various features like tabs for history, contacts, and accounts.
	- Added a new `ImageEdit` component for uploading profile pictures.
	- Implemented a `TextEdit` component for inline text editing.
	- Introduced a `MobileTableComponent` for responsive profile data display.
	- Added an `OptionsComponent` for rendering customizable action menus.
	- Introduced a `TableComponent` for structured profile data presentation.
	- Added a `Tabs` component for improved navigation within the profile interface.
	- Introduced new icons and enhanced the `Icon` component to support click events.

- **Improvements**
	- Updated metadata descriptions across multiple pages to reflect a shift in focus to "Text Tokens."
	- Introduced utility functions for date formatting and icon name retrieval.
	- Refactored the `Dashboard` component to improve state management and data fetching.
	- Improved the `useBalance` hook to reduce UI flickering during balance retrieval.

- **Bug Fixes**
	- Fixed navigational links to direct users to the profile instead of the dashboard.
	- Improved error handling and user feedback mechanisms for various user actions.

- **Style**
	- Added custom skeleton loading styles for better visual feedback.
	- Updated component styles for improved consistency and user experience across the application.
	- Enhanced loading component styles to ensure default sizes and flexible class naming.
	- Introduced a new color palette with the addition of a pink shade for enhanced theming.

### CODE REVIEW

File: src/components/Profile/Components/MobileTableComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    tertiaryText,
+    quaternaryText,
+    type,
+    avatar,
+    dashboardItem,
+    address,
+}: interfaces.IProfileTableData) => {
+    const [modalVisible, setModalVisible] = useState(false)
+    const router = useRouter()
+    const handleSendToAddress = useCallback(
```
Comment (borcherd) [inline src/components/Profile/Components/MobileTableComponent.tsx]: Can this not be done with a <Link> @pawell24 ?

File: src/components/Profile/Components/TableComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    currentPage,
+    itemsPerPage,
+}: {
+    data: interfaces.IProfileTableData[]
+    selectedTab: 'contacts' | 'history' | 'accounts' | undefined
+    currentPage: number
+    itemsPerPage: number
+}) => {
+    const router = useRouter()
+    const handleSendToAddress = useCallback(
```
Comment (borcherd) [inline src/components/Profile/Components/TableComponent.tsx:23]: Same here

Comment (pawell24) [reply src/components/Profile/Components/TableComponent.tsx:23]: I think If we want use Link component in Options component we need to change structure of Options component. I don't think it's worth the benefits we'll get. The structure is different than in the mobile view. What do you think? @borcherd

File: src/components/Profile/Components/MobileTableComponent.tsx (hunk 1 above)
Comment (pawell24) [reply src/components/Profile/Components/MobileTableComponent.tsx]: Fixed

File: src/components/Global/ImageEdit/index.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    const toast = useToast({
+        position: 'bottom-right',
+        duration: 5000,
+        isClosable: true,
+        icon: '🥜',
+    })
+
+    const handleImageUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const file = event.target.files ? event.target.files[0] : null
+        if (file && file.type.startsWith('image/')) {
```
[the remark this answers is not in the document]
Comment (pawell24) [reply src/components/Global/ImageEdit/index.tsx]: It's worth to add optional chaining @borcherd

File: src/app/api/peanut/user/get-jwt-token/route.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+        const data = await response.json()
+        const token = data.token
+
+        // Set the JWT token in a cookie, nextjs requires to do this serverside
+        cookies().set('jwt-token', token, {
+            httpOnly: true,
+            path: '/',
+            sameSite: 'strict',
+        })
```
[the remark this answers is not in the document]
Comment (pawell24) [reply src/app/api/peanut/user/get-jwt-token/route.ts]: Let's take a look about secure attribute in cookies @borcherd

**Review by pawell24 — APPROVED** [review]

**Pull request merged**